### PR TITLE
Re roll 6279

### PR DESCRIPTION
**Pull request by esc**
As title: re-roll https://github.com/numba/numba/pull/6279 due to git issues.

### CODE REVIEW

**Comment by esc** [comment]
@stuartarchibald @sklam @rishabhvarshney14 would you be able to check if I cherry-picked everything appropriately? It's probably enough to look at the final diff of this PR.

File: numba/tests/test_np_functions.py [hunk 1]
```diff
@@ -3906,7 +3906,11 @@ def test_asarray_chkfinite(self):
             # 3x2 (with higher order broadcasting)
             (
                 np.arange(36).reshape(6, 2, 3),
-            )
+            ),
+            # (
```
Comment (rishabhvarshney14) [inline numba/tests/test_np_functions.py:3910]: I forgot to remove this comments here and made another commit after this to remove this

Comment (esc) [reply numba/tests/test_np_functions.py]: Yeah, I think that commit was lost, I made another one to fix it, thanks!

**Review by sklam — APPROVED** [review]

**Comment by esc** [comment]
@rishabhvarshney14 finally, congrats are in order! Looks like this one made it just in the nick of time for the 0.52.0RC1 -- be on the lookout for a mention in the release notes!

**Comment by esc** [comment]
@sklam thanks for approval and merge!

**Pull request merged**